### DISCUSSION TITLE
quick/scaffold-index

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#45] [General]
# quick/scaffold-index

本章节详细介绍如何使用GoFrame框架的脚手架来构建一个简单且完整的业务项目。内容涵盖框架工具的安装与验证、创建工程项目模板、项目模板的运行以及如何升级框架版本等重要步骤。项目模板默认支持HTTP Web Server，并提供接口文档展示与Swagger页面查看功能。

https://goframe.org/quick/scaffold-index

**Comment by Mobs-demacia** [comment]
新官网支持!  能否添加点击图片放大查看的功能.

**Reply by gqcn** [reply]
感谢建议，已支持。

**Comment by xiaotangwang** [comment]
这个是什么原因？修改了 service.yaml 和 configmap.yaml 中的targetPort : 8989 不生效。
[FATA] net.Listen address ":8000" failed: listen tcp :8000: bind: An attempt was made to access a socket in a way forbidden by its access permissions.
1. net.Listen address ":8000" failed
2. listen tcp :8000: bind: An attempt was made to access a socket in a way forbidden by its access permissions.

**Reply by xiaotangwang** [reply]
这个是使用 gf init demo 自动生成的代码

**Reply by gqcn** [reply]
看起来似乎你当前的用户无法监听所有ip或者该端口，你可以尝试修改监听地址为`127.0.0.1:8000`试试，比如：
```yaml
server:
  address: "127.0.0.1:8000"
```

**Reply by hailaz** [reply]
端口已经被占用了

**Comment by wangxiaofan777** [comment]
这个框架的认证功能有没有呀？

**Comment by s72267534-stack** [comment]
got the project zip data failed: parse "https://gfcdn.johng.cn/cli/project/zip?<html>\r\n<head><title>404 Not Found</title></head>\r\n<body>\r\n<center><h1>404 Not Found</h1></center>\r\n<hr><center>nginx/1.18.0 (Ubuntu)</center>\r\n</body>\r\n</html>\r\n": net/url: invalid control character in URL

**Comment by themagicaster-create** [comment]
demo\api\hello\hello.go和demo\internal\controller\hello\hello_v1_hello.go里没有导入v1包，注意一下

**Comment by hybfly** [comment]
我运行的时候出现如下错误，请问是什么原因？
watching 1 paths
build: main.go
go build -o .\main.exe  main.go
build error:
# demo/api/hello
api\hello\hello.go:12:34: undefined: v1
exit status 1

**Reply by hailaz** [reply]
go mod tidy 一下，应该是没import
这个是正常的例子
https://github.com/gogf/template-single/blob/main/api/hello/hello.go#L10

**Reply by hybfly** [reply]
谢谢！已解决！

**Comment by liqianwithship** [comment]
不同项目想使用不同的 gf 版本，如何实现呢？

**Reply by hailaz** [reply]
可以手动将下载的二进制重命名，例如gf29 gf210

又或者 go run github.com/gogf/gf/cmd/gf/v2@latest -v这样直接调用
